src/inspector_agent.cc

```diff
@@ -606,11 +623,18 @@ class NodeInspectorClient : public V8InspectorClient {
   bool waiting_for_io_shutdown_ = false;
   // Allows accessing Inspector from non-main threads
   std::unique_ptr<MainThreadInterface> interface_;
+  std::shared_ptr<WorkerManager> worker_manager_;
 };
 
 Agent::Agent(Environment* env) : parent_env_(env) {}
 
-Agent::~Agent() = default;
+Agent::~Agent() {
+  if (start_io_thread_async.data == this) {
+    start_io_thread_async.data = nullptr;
+    // This is global, will never get freed
+    uv_close(reinterpret_cast<uv_handle_t*>(&start_io_thread_async), nullptr);
```

Comment:
This is interesting – why is `start_io_thread_async` global? That doesn’t seem quite right (or thread-safe, for that matter). These things should probably per-Environment or per-Isolate?

Reply:
It is used by a signal handler.